src/harness/testcases/WINNF_FT_S_MCP_testcase.py

```diff
@@ -306,3 +311,214 @@ def test_WINNF_FT_S_MCP_1(self, config_filename):
+          self._sas_admin.InjectWisp({'record': gwpz_record})
+
+      if 'escRecords' in protected_entity_records:
+        for esc_record in protected_entity_records['escRecords']:
+          self._sas_admin.InjectEscSensorDataRecord({'record': esc_record})
+
+      if 'ppaRecords' in protected_entity_records:
+        for ppa_record in protected_entity_records['ppaRecords']:
+          self._sas_admin.InjectZoneData({'record': ppa_record})
+
+      # Step 6,7 : Creating FAD Object and Pull FAD records from SAS UUT
+      if len(iteration_content['sasTestHarnessData']) > 0:
+        fad_uut_object = getFullActivityDumpSasUut(self._sas,
+                                                   self._sas_admin,
+                                                   sas_uut_cert,
+                                                   sas_uut_key)
+
+        # Pull FAD from SAS Test Harnesses and FAD objects are created for each SAS Test Harnesses
+        for index, test_harness in enumerate(sas_test_harness_objects):
+          fad_test_harnesses_objects.append(getFullActivityDumpSasTestHarness(
+                                  test_harness.getSasTestHarnessInterface(),
+                                  config['sasTestHarnessConfigs'][index]['serverCert'],
+                                  config['sasTestHarnessConfigs'][index]['serverKey']
+                              )
+          )
+
+      # Step 8 : Trigger CPAS and wait until completion
+      self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+      # Step 9 : Invoke IAP reference model
+      # Call Pre IAP model
+      # TODO: To invoke pre-iap model
+
+      # Call IAP reference model
+      # TODO: To invoke IAP reference model
+
+      # Step 10 : DP Test Harness Register N(2,k)CBSDs with SAS UUT
+      # Use DP objects to get CBSD registered
+      for domain_proxy_object in domain_proxy_objects:
+        for index, cbsdRequestsWithDomainProxys in enumerate(iteration_content['cbsdRequestsWithDomainProxies']):
+          registration_requests = cbsdRequestsWithDomainProxys['registrationRequests']
+          grant_requests = cbsdRequestsWithDomainProxys['grantRequests']
+
+          # domain_proxy_object.initialize triggers the registration and grant for the records passes as parameters
+          domain_proxy_object.initialize(registration_requests, grant_requests)
+
+          # Step 11 : Send heartbeat request
+          domain_proxy_object.heartbeatRequestForAllGrants()
+
+          # Get the list of CBSDs
+          cbsd_agg_list.append(domain_proxy_object.getCbsdsWithAtLeastOneAuthorizedGrant())
+
+      # Register individual Cbsds
+      for cbsd_record in iteration_content['cbsdRecords']:
+        cbsd_ids, grant_ids = self.assertRegisteredAndGranted([cbsd_record['registrationRequest']],
+                                                              [cbsd_record['grantRequest']])
+        cbsd_agg_list.append(cbsd_ids)
```

Comment:
getCbsdsWithAtLeastOneAuthorizedGrant returns Cbsd objects, assertRegisteredAndGranted returns CBSD ids (I think this is a string). So currently 2 different data types are in this list, the CBSD id one probably doesn't have sufficient information for this test.